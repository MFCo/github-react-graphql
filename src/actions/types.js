import keyMirror from 'key-mirror';

export default keyMirror({
  UPDATE_USER: null,
  UPDATE_REPOSITORIES: null,
  ADD_REPOSITORIES: null,
  UPDATE_DIRECTORY: null,
  UPDATE_REPOSITORY: null
});