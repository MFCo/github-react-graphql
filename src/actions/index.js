import ACTIONS from './types';
import { createActions } from 'redux-actions';

export default createActions(...Object.keys(ACTIONS));