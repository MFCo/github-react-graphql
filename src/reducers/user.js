import { handleActions } from 'redux-actions';

const notProvided = (data) => `no ${data} provided`;

const defaultState = {
  avatar: '',
  name: notProvided('name'),
  user: notProvided('user'),
  bio: notProvided('bio'),
  company: notProvided('company'),
  location: notProvided('location'),
  mail: notProvided('e-mail'),
  website: notProvided('website')
};

export default handleActions({
  UPDATE_USER: (state, { payload: { newAvatar, newName, newUser, newBio, newCompany, newLocation, newMail, newWebsite } }) => {
    return {
      avatar: newAvatar,
      name: newName,
      user: newUser,
      bio: newBio,
      company: newCompany,
      location: newLocation,
      mail: newMail,
      website: newWebsite
    }
  }
}, defaultState);