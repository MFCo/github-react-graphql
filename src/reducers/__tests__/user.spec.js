import user from '../user';
import actions from '../../actions';

it('should return the initial state', () => {
  expect(user(undefined, {})).toMatchSnapshot()
})

it('should handle UPDATE_USER', () => {
  expect(user(undefined, actions.updateUser({
    newName: 'Jhon Doe',
    newUser: 'JDoe',
    newAvatar: 'image.jpg',
    newBio: 'hi!',
    newCompany: 'company',
    newLocation: 'London, Spain',
    newMail: 'gmail@jhondoe.com',
    newWebsite: 'com.ar.jhon'
  }))).toMatchSnapshot()
})