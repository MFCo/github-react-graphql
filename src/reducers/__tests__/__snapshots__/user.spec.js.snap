// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle UPDATE_LIST 1`] = `
Object {
  "avatar": "image.jpg",
  "bio": "hi!",
  "company": "company",
  "location": "London, Spain",
  "mail": "gmail@jhondoe.com",
  "name": "Jhon Doe",
  "user": "JDoe",
  "website": "com.ar.jhon",
}
`;

exports[`should return the initial state 1`] = `
Object {
  "avatar": "",
  "bio": "no bio provided",
  "company": "no company provided",
  "location": "no location provided",
  "mail": "no e-mail provided",
  "name": "no name provided",
  "user": "no user provided",
  "website": "no website provided",
}
`;
